app/controllers/hyrax/handles_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Hyrax
+  class HandlesController < ApplicationController
+    def handle_show
+      if params[:handle_id] != "1957"
+        puts "Handle ID does not match with Scholars Archive Handle ID."
+      elsif params[:id].length != 5
+        puts "ID is not the right length for Scholars Archive Handle ID."
+      end
+
+      solr_doc = query_solr_for_work(construct_handle_url(params[:id]))
+      work = query_fedora_for_work(solr_doc.first[:id], solr_doc.first[:human_readable_type_tesim].first.constantize)
+      redirect_to "/concern/#{work.class.to_s.downcase}s/#{work.id}"
+    end
+
+    def handle_download
+    end
+
+    private
+      def construct_handle_url(handle_code)
+        "hdl.handle.net/1954/" + handle_code
+      end
+
+      def query_fedora_for_work(id, work_type)
+        work_type.find(id)
+      end
+
+      def query_solr_for_work(handle)
+        #Query solr.
+        ActiveFedora::SolrService.query("replaces_tesim:#{handle}", :rows => 1000000)
```

Comment:
does `replaces_tesim` have the full "http://hdl.handle.net...." or just the portion you're passing in as `handle`.. AFAICT it should include the `http://`